File: NapakalakiRuby/lib/death_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "numeric_bad_consequence.rb"

class DeathBadConsequence < NumericBadConsequence
  
  def initialize(t, d)
    super(t, Player::MAXLEVEL, BadConsequence::MAXTREASURES, 
      BadConsequence::MAXTREASURES)
    @death = d
  end
  
  # Getters
  def get_death
    @death
  end
  
  # Setters
  def set_death(d)
    @death = d
  end
  
  # Utils
  def is_empty
    super && !@death
  end
  
  def to_s
    super + ", Death = #{@death}"
  end
  
end

File: Mis Prácticas/Napakalaki/src/napakalaki/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author fran
 */
public class DeathBadConsequence extends NumericBadConsequence{
    
    // Variables
    private boolean death;
    
    // Constructor
    public DeathBadConsequence(String text, boolean death){
        super(text, Player.MAXLEVEL, BadConsequence.MAXTREASURES, 
                BadConsequence.MAXTREASURES);
        setDeath(death);
    }
    
    // Getters
    public boolean getDeath(){
        return death;
    }
    
    // Setters
    public void setDeath(boolean death){
        this.death = death;
    }
    
    @Override
    public String toString(){
        
        return super.toString()+ ", Death = " + death;
        
    }
    
}
